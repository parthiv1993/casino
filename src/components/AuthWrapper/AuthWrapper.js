import { useContext } from 'react';
import PropTypes from 'prop-types';
import { Navigate, useLocation } from 'react-router-dom';
import LoginContext from '../../contexts/login/LoginContext';

const AuthWrapper = ({ children }) => {
  const { user } = useContext(LoginContext);
  let location = useLocation();

  if (user) {
    return children;
  }

  return <Navigate to="/login" state={{ from: location }} replace />;
};

AuthWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default AuthWrapper;
