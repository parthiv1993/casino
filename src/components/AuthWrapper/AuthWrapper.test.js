/* eslint-disable react/display-name */
import { render } from '@testing-library/react';
import LoginContext from '../../contexts/login/LoginContext';
import AuthWrapper from './AuthWrapper';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Navigate: (props) => <div props={JSON.stringify(props)} />,
  useLocation: () => 'location'
}));

const dummyUser = {
  name: 'mockUser',
  event: 'event String'
};

describe('AuthWrapper', () => {
  it('should render children if user is set in LoginContext', () => {
    const { container } = render(
      <LoginContext.Provider value={{ user: dummyUser }}>
        <AuthWrapper>This is dummy text</AuthWrapper>
      </LoginContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render navigate when user is not present', () => {
    const { container } = render(
      <LoginContext.Provider value={{ user: null }}>
        <AuthWrapper>This is dummy text</AuthWrapper>
      </LoginContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
