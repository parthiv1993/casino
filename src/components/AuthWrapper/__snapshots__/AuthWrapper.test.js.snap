// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthWrapper should render children if user is set in LoginContext 1`] = `
<div>
  This is dummy text
</div>
`;

exports[`AuthWrapper should render navigate when user is not present 1`] = `
<div>
  <div
    props="{\\"to\\":\\"/login\\",\\"state\\":{\\"from\\":\\"location\\"},\\"replace\\":true}"
  />
</div>
`;
